File: app/src/main/java/example/simple/itemdecor/linear/LinearHorizontalItemBinder.java
package example.simple.itemdecor.linear;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import me.drakeet.multitype.ItemViewBinder;
import me.simple.itemdecor.demo.R;

public class LinearHorizontalItemBinder extends ItemViewBinder<LinearItemBean, LinearHorizontalItemBinder.VH> {

    @NonNull
    @Override
    protected VH onCreateViewHolder(@NonNull LayoutInflater inflater, @NonNull ViewGroup parent) {
        return new VH(inflater.inflate(R.layout.item_linear_horizontal, parent, false));
    }

    @Override
    protected void onBindViewHolder(@NonNull VH holder, @NonNull LinearItemBean item) {

    }

    class VH extends RecyclerView.ViewHolder {

        public VH(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/example/simple/itemdecor/linear/LinearHorizontalItemBinder.kt
package example.simple.itemdecor.linear

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import example.simple.itemdecor.R
import me.drakeet.multitype.ItemViewBinder

class LinearHorizontalItemBinder : ItemViewBinder<LinearItemBean, LinearHorizontalItemBinder.VH>() {
    override fun onCreateViewHolder(inflater: LayoutInflater, parent: ViewGroup): VH {
        return VH(inflater.inflate(R.layout.item_linear_horizontal, parent, false))
    }

    override fun onBindViewHolder(holder: VH, item: LinearItemBean) {}
    inner class VH(itemView: View) : ViewHolder(itemView)
}